app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorDrawerActivity.kt

```diff
@@ -1,11 +1,104 @@
+        val drawerLayout: DrawerLayout = binding.drawerLayout
+        val navView: NavigationView = binding.navView
+        val navController = findNavController(R.id.nav_host_fragment_content_drawer)
+        appBarConfiguration = AppBarConfiguration(
+            setOf(R.id.nav_home),
+            drawerLayout
+        )
+        setupActionBarWithNavController(navController, appBarConfiguration)
+        navView.setupWithNavController(navController)
+        val navigationView1 = findViewById<NavigationView>(R.id.nav_view)
+        val headerview = navigationView1.getHeaderView(0)
+        userNames = headerview.findViewById(R.id.donertv)
+        userNames.text = "" + FirebaseAuth.getInstance().currentUser?.displayName
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel =
+                NotificationChannel("myCh", "Channel1", NotificationManager.IMPORTANCE_HIGH)
+            val manager = getSystemService(NotificationManager::class.java)
+            manager.createNotificationChannel(channel)
+        }
+        FirebaseMessaging.getInstance()
+            .subscribeToTopic("" + FirebaseAuth.getInstance().currentUser?.uid)
+        navigationView1.menu.getItem(1).setOnMenuItemClickListener {
+            drawerLayout.closeDrawers()
+            val i = Intent(this@DonorDrawerActivity, PostsActivity::class.java)
+            startActivity(i)
+            false
+        }
+        navigationView1.menu.getItem(2).setOnMenuItemClickListener {
+            drawerLayout.closeDrawers()
+            val i = Intent(this@DonorDrawerActivity, RequestActivity::class.java)
+            i.putExtra("user", "")
+            startActivity(i)
+            false
+        }
+        navigationView1.menu.getItem(3).setOnMenuItemClickListener {
+            drawerLayout.closeDrawers()
+            val i = Intent(this@DonorDrawerActivity, UserActivity::class.java)
+            startActivity(i)
+            false
+        }
+        navigationView1.menu.getItem(4).setOnMenuItemClickListener {
+            val i = Intent(this@DonorDrawerActivity, DonorDrawerActivity::class.java)
+            drawerLayout.closeDrawers()
+            startActivity(i)
+            false
+        }
+        navigationView1.menu.getItem(5).setOnMenuItemClickListener {
+            drawerLayout.closeDrawers()
+            circularProgressIndicator.visibility = View.VISIBLE
+            circularProgressIndicator.visibility = View.GONE
+            FirebaseAuth.getInstance().signOut()
+            val i = Intent(this, HomescreenActivity::class.java)
+            startActivity(i)
+            finishAffinity()
+            false
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.drawer, menu)
+        return true
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        val navController = findNavController(R.id.nav_host_fragment_content_drawer)
```

Comment:
this code isn't required if you aren't using multiple fragments in the navigation graph